### ISSUE TITLE
Array length and iteration in reverse?

### ISSUE THREAD

**Issue by SodaPopinski** [untoldwind/KontrolSystem2#15]
Is it possible to get the length of an array?

Is it possible to do a For loop iterating backwards?

For some background, I wanted to iterate through an array backwards, but failing how to figure out how to do that, I planned to do a While true loop, counting backwards, but I couldn't figure out what index number to start with, since I couldn't get the length of the array.
On a related note, what is the exact syntax to iterate numerically in a For loop in reverse?  I was going to try variations based on C# code.

**Comment by untoldwind** [comment]
The length of an array is just `[1, 2, 3].length` (probably missing in the documentation as well).

Iterating backward is indeed clunky at the moment, i.e.
```
for(i in 1...(arr.length)) {
   let element = arr[arr.length - i]
}
```
should do the trick, but that not really elegant.
I think I will add a `reverse` property to array and allow ranges to go in reverse as well.

**Comment by untoldwind** [comment]
Documented all the methods on array: https://kontrolsystem2.readthedocs.io/en/latest/language/special_types/array.html

`.reverse()` should now be available in 0.1.8

(Note: It is a method, not a property because it is really hard to deal with generic types)

**Comment by untoldwind** [comment]
I am closing this for now
